### ISSUE TITLE
VS crashes if you rename an F# symbol where it is at the top of a file

### ISSUE THREAD

**Issue by cartermp** [Microsoft/visualfsharp#4655]
Using 15.6.4 and latest master installed into it, attempt to rename `Record` in the following code:

```fsharp
type Record = { IntVal: int; StringVal: string }

[<AbstractClass>]
type AbstractFoo<'T>() =
    abstract member Foo: 'T -> 'T

type C() =
    inherit AbstractFoo<Record>()
    override __.Foo(r) = { r with IntVal = 12 }
```

After typing the first character, the inline rename service will crash, followed by VS crashing. 

This is the stack trace in the activity log:

```
System.NullReferenceException: Object reference not set to an instance of an object. at 
Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameSession.CommitCore(IWaitContext waitContext, Boolean previewChanges) at 
Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameSession.<>c__DisplayClass71_0.<Commit>b__0(IWaitContext waitContext) at 
Microsoft.VisualStudio.LanguageServices.Implementation.Utilities.VisualStudioWaitIndicator.Wait(String title, String message, Boolean allowCancel, Boolean showProgress, Action`1 action) at 
Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameSession.Commit(Boolean previewChanges) at 
Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.RenameCommandHandler.CommitIfActive(EditorCommandArgs args) at 
Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.RenameCommandHandler.CommitIfActiveAndCallNextHandler(EditorCommandArgs args, Action nextHandler) at 
Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.RenameCommandHandler.HandlePossibleTypingCommand(EditorCommandArgs args, Action nextHandler, Action`1 actionIfInsideActiveSpan) at 
Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.RenameCommandHandler.ExecuteCommand(TypeCharCommandArgs args, Action nextHandler, CommandExecutionContext context) at 
Microsoft.VisualStudio.Commanding.CommandHandlerExtensions.ExecuteCommand[T](ICommandHandler commandHandler, T args, Action nextCommandHandler, CommandExecutionContext executionContext) at 
Microsoft.VisualStudio.UI.Text.Commanding.Implementation.EditorCommandHandlerService.<>c__DisplayClass14_1`1.<Execute>b__1() at 
Microsoft.VisualStudio.Text.Utilities.GuardedOperations.CallExtensionPoint(Object errorSource, Action call) --- End of stack trace from previous location where exception was thrown --- at 
Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
```

Currently this makes it impossible to test out a fix for #4231 😢

**Comment by abelbraaksma** [comment]
Some observations, in case it helps:

This is not caused by the latest master, at the very least the bug has been out for a little while longer. I haven't updated to latest master, but I can repro in an instance of VS2017 15.6.4 and VS2017 15.7.0 Preview 2 (which means, I believe, that Preview 2 is "installed into" 15.6.4. as well wrt F# services and compiler).

![image](https://user-images.githubusercontent.com/16015770/38175661-684f7a82-35e0-11e8-93ef-74d6851e098d.png)

I also noted that the lookup-screen says "will replace 0 references in 0 files", perhaps that is a hint?

![image](https://user-images.githubusercontent.com/16015770/38175677-b1e8e160-35e0-11e8-8507-ffcab88616c9.png)

Even more worrying, when I put back focus on the editor while the rename-box is still visible top-right, I get this ("An exception has been encountered"):

![image](https://user-images.githubusercontent.com/16015770/38175693-dc6b21e6-35e0-11e8-9ec1-1b2c5ba80f2e.png)

And, still more surprising, even when I *don't apply the renaming* (hit cancel or the X in top-right of rename window), it still crashes!

**Comment by dsyme** [comment]
@cartermp @abelbraaksma I don't get a crash, but I do get a double-rename of he generic parameter:

e.g.
```
type Record = { IntVal: int; StringVal: string }
...
    inherit AbstractFoo<Record>()
```
becomes
```
type AA = { IntVal: int; StringVal: string }
...
    inherit AbstractFoo<AAAA>()
```

**Comment by dsyme** [comment]
I fixed the double renaming in #4810

**Comment by cartermp** [comment]
@dsyme Did you test this with just 15.6.x, or with 15.6.x and a master VSIX built in, or 15.7? The crash occured for me in the latter two setups, but not "normal" 15.6.x.

**Comment by dsyme** [comment]
Just 15.6.x  plus F5 launch of Visualfsharp. sln

**Comment by buybackoff** [comment]
Probably the same issue, I see an error right after I press F2:
![image](https://user-images.githubusercontent.com/489852/40057696-abcfc4e4-5857-11e8-81bc-8c251918d815.png)
Rename doesn't happen, but the 'rename apply' window is always open when the editor windows is in focus:
![image](https://user-images.githubusercontent.com/489852/40057714-b7134394-5857-11e8-85fc-315e18b8d129.png)
Same place and machine config (VS 15.7) as in #4209 with today's (May 15) nightly build.

**Comment by chillitom** [comment]
Tested out the original example a couple of times on 15.7.1 and once got the "An exception has been encountered" message and once got the double renaming (now fixed in nightlies).

**Comment by chillitom** [comment]
The double rename also occurs if I rename AbstractFoo.  I've repeated the test roughly 25 times and haven't been able to recreate the exception.  Perhaps requires a solution that has been open/churning for a while to trigger the conditions for this crash.

**Comment by chillitom** [comment]
Error from ActivityLog.xml
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameSession.CommitCore(IWaitContext waitContext, Boolean previewChanges)
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameSession.<>c__DisplayClass71_0.<Commit>b__0(IWaitContext waitContext)
   at Microsoft.VisualStudio.LanguageServices.Implementation.Utilities.VisualStudioWaitIndicator.Wait(String title, String message, Boolean allowCancel, Boolean showProgress, Action`1 action)
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameSession.Commit(Boolean previewChanges)
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.RenameCommandHandler.ExecuteCommand(ReturnKeyCommandArgs args, CommandExecutionContext context)
   at Microsoft.VisualStudio.Commanding.CommandHandlerExtensions.ExecuteCommand[T](ICommandHandler commandHandler, T args, Action nextCommandHandler, CommandExecutionContext executionContext)
   at Microsoft.VisualStudio.UI.Text.Commanding.Implementation.EditorCommandHandlerService.<>c__DisplayClass14_1`1.<Execute>b__1()
   at Microsoft.VisualStudio.Text.Utilities.GuardedOperations.CallExtensionPoint(Object errorSource, Action call)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
```

**Comment by chillitom** [comment]
That sounds plausible.  The odd thing was I copy pasted the example into the foot of a file in a lagre F# solution and immediately got the exception on first attempt.  After restart I've not been able to trigger it again.  I'd hoped I'd be able to attach a debugger but wasn't prompted as sometimes happens, I think because the main VS process survived the crash.  The rename box became unusable after the error message and couldn't be dismissed.

**Comment by cartermp** [comment]
This is fixed in #4939
